Classes/Outlet/AbstractOutlet.php

```diff
@@ -132,10 +157,15 @@ public function addPipeOut(PipeInterface $pipe)
      */
     public function fill($data)
     {
+        $this->validate($data);
         foreach ($this->pipesIn as $pipe) {
+            if ($pipe instanceof ViewAwarePipeInterface) {
```

Comment:
This will either always be true - or we should throw an error if pipe is not the right interface.